src/configs/FirebaseConfig.js

```diff
@@ -14,20 +16,41 @@ import {
 import {getUserId} from '../utils/users';
 import {isAuthorFollowingMe} from '../service/post';
 
+const DEEPLINK_POST_REGEX = /helio\.social\/post\/[0-9a-fA-F\-]{36}/;
+const DEEPLINK_COMMUNITY_REGEX = /helio\.social\/c\/[a-zA-Z0-9]+/;
+const DEEPLINK_USER_REGEX = /helio\.social\/[a-zA-Z0-9]+/;
+
 const FirebaseConfig = () => {
   const navigation = useNavigation();
+  const {signedProfileId, profile} = useUserAuthHook();
 
   React.useEffect(() => {
     const unsubscribe = dynamicLinks().onLink(parseDynamicLink);
-    return () => unsubscribe();
+    const onForegroundEvent = async (event) => {
+      parseDeepLink(event?.url);
+    };
+
+    Linking.addEventListener('url', onForegroundEvent);
+    return () => {
+      unsubscribe();
+      Linking.removeEventListener('url', onForegroundEvent);
+    };
   }, []);
 
+  const parseDeepLink = async (deepLinkUrl) => {
+    if (DEEPLINK_POST_REGEX.test(deepLinkUrl)) return handleDeepLinkPost(deepLinkUrl);
+    if (DEEPLINK_COMMUNITY_REGEX.test(deepLinkUrl)) return handleDeepLinkCommunity(deepLinkUrl);
+    if (DEEPLINK_USER_REGEX.test(deepLinkUrl)) return handleDeepLinkUser(deepLinkUrl);
+  };
+
   /**
    *
    * @param {FirebaseDynamicLinksTypes.DynamicLink} dynamicLink
    */
   const parseDynamicLink = async (dynamicLink) => {
-    console.log('called dynamic link', dynamicLink);
+    setTimeout(() => {
+      Alert.alert('Dynamic Link from dynamicLink', dynamicLink?.url);
+    }, 1000);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The regular expressions used to match deep links are not anchored. This means they can match anywhere in the string, which could lead to unexpected matches if the URL contains additional paths or parameters. Consider anchoring your regular expressions with `^` at the start and `$` at the end to ensure they match the entire string.

Also, it's a good practice to handle errors that might occur during the execution of async functions. In the `parseDeepLink` function, you should add error handling logic to catch any potential exceptions that might be thrown when calling `handleDeepLinkPost`, `handleDeepLinkCommunity`, and `handleDeepLinkUser`.

```diff
- const DEEPLINK_POST_REGEX = /helio\.social\/post\/[0-9a-fA-F\-]{36}/;
- const DEEPLINK_COMMUNITY_REGEX = /helio\.social\/c\/[a-zA-Z0-9]+/;
- const DEEPLINK_USER_REGEX = /helio\.social\/[a-zA-Z0-9]+/;
+ const DEEPLINK_POST_REGEX = /^helio\.social\/post\/[0-9a-fA-F\-]{36}$/;
+ const DEEPLINK_COMMUNITY_REGEX = /^helio\.social\/c\/[a-zA-Z0-9]+$/;
+ const DEEPLINK_USER_REGEX = /^helio\.social\/[a-zA-Z0-9]+$/;

  const parseDeepLink = async (deepLinkUrl) => {
    try {
      if (DEEPLINK_POST_REGEX.test(deepLinkUrl)) return await handleDeepLinkPost(deepLinkUrl);
      if (DEEPLINK_COMMUNITY_REGEX.test(deepLinkUrl)) return await handleDeepLinkCommunity(deepLinkUrl);
      if (DEEPLINK_USER_REGEX.test(deepLinkUrl)) return await handleDeepLinkUser(deepLinkUrl);
    } catch (error) {
      console.error('Error parsing deep link:', error);
    }
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->